cellular_configs/v1alpha1/commoncoreranconfig_types.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CommonCoreRANStatus defines the observed state of CommonCoreRAN
+type CommonCoreRANConfigStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type CommonCoreRANConfig struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   CommonCoreRANConfigSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status CommonCoreRANConfigStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// CommonCoreRANConfigList contains a list of CommonCoreRANConfigs
+type CommonCoreRANConfigList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []CommonCoreRANConfig `json:"items" yaml:"items"`
+}
+
+// CommonCoreRANConfigSpec defines the characteristics of a deployment of a network function
+type CommonCoreRANConfigSpec struct {
+	// plmnInfo defines the list of PLMNs to configure for core NFs
+	PlmnInfo []PlmnInfoConfig `json:"plmnInfo,omitempty" yaml:"plmnInfo,omitempty"`
+	// communicationModel defines the communcation strategy between core components, using NRF or without NRF
+	// +optional
+	CommunicationModel []CommunicationModelConfig `json:"communicationModel,omitempty" yaml:"communicationModel,omitempty"`
```

Comment:
When and why do we have multiple communication models? What is the scenario ?

Reply:
I think for OAI, UPF talks to NRF, whereas in free5gc, UPF doesn't even know what a NRF is

Reply:
@henderiw @s3wong not really again this comes from 3gpp yang models. In OAI also I can do something similar to free5GC. Communication model depicts if NFs communicate with NRF (5G service based architecture) or not (5G reference based architecture). 

If Nephio will only deploy 5G Service Based Architecture then I can remove this parameter. I just wanted to have a way to deploy both architectures if possible and if needed by someone.

Reply:
My point is for a given deployment do you need more then 1 ?

Reply:
@henderiw For a given deployment there is only one communication model either 
with NRF or without NRF.